src/server.h

```diff
@@ -83,6 +84,7 @@ typedef long long ustime_t; /* microsecond time type. */
 
 #define VALKEYMODULE_CORE 1
 typedef struct serverObject robj;
+typedef struct serverObject valkey;
```

Comment:
I wanted to distinguish an object that has a key from one that doesn't, but I agree this doesn't add any type safety.

We do something similar in networking.c when a client is prepared for write. We cast it like `(writePreparedClient *)c` but `writePreparedClient` is not a typedef. It's just a struct without a definition, so the only way to use it is casting.

Would it make sense to do something like that here?

Otherwise I'll just remove the valkey typedef.

Reply:
Dropped the valkey typedef.